# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IDBObjectStore.cpp
# Commit: 43dbad71664b
# Full Hash: 43dbad71664bb2c5779b54decd5133db076ec6a0
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-11-08 21:36:09
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Replace pseudo-move constructor of StructuredCloneReadInfo by explicit DeserializeStructuredCloneReadInfo function. r=ttung,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46593
# ==============================================================================

diff -r 19188ca45113 -r 43dbad71664b dom/indexedDB/IDBObjectStore.cpp
--- a/dom/indexedDB/IDBObjectStore.cpp	Fri Nov 08 14:28:39 2019 +0000
+++ b/dom/indexedDB/IDBObjectStore.cpp	Fri Nov 08 13:22:08 2019 +0000
@@ -273,9 +273,9 @@
       return false;
     }
 
-    StructuredCloneFile* const newFile = cloneWriteInfo->mFiles.AppendElement();
-    newFile->mMutableFile = mutableFile;
-    newFile->mType = StructuredCloneFile::eMutableFile;
+    const DebugOnly<StructuredCloneFile*> newFile =
+        cloneWriteInfo->mFiles.EmplaceBack(mutableFile);
+    MOZ_ASSERT(newFile);
 
     return true;
   }
@@ -337,10 +337,9 @@
         }
       }
 
-      StructuredCloneFile* const newFile =
-          cloneWriteInfo->mFiles.AppendElement();
-      newFile->mBlob = blob;
-      newFile->mType = StructuredCloneFile::eBlob;
+      const DebugOnly<StructuredCloneFile*> newFile =
+          cloneWriteInfo->mFiles.EmplaceBack(StructuredCloneFile::eBlob, blob);
+      MOZ_ASSERT(newFile);
 
       return true;
     }
@@ -381,9 +380,9 @@
         return false;
       }
 
-      StructuredCloneFile* const newFile = cloneInfo->mFiles.AppendElement();
-      newFile->mBlob = blob;
-      newFile->mType = StructuredCloneFile::eBlob;
+      const DebugOnly<StructuredCloneFile*> newFile =
+          cloneInfo->mFiles.EmplaceBack(StructuredCloneFile::eBlob, blob);
+      MOZ_ASSERT(newFile);
 
       return true;
     }
@@ -404,9 +403,9 @@
         return false;
       }
 
-      StructuredCloneFile* const newFile = cloneInfo->mFiles.AppendElement();
-      newFile->mMutableFile = mutableFile;
-      newFile->mType = StructuredCloneFile::eMutableFile;
+      const DebugOnly<StructuredCloneFile*> newFile =
+          cloneInfo->mFiles.EmplaceBack(mutableFile);
+      MOZ_ASSERT(newFile);
 
       return true;
     }